client/src/models/opportunityfilterproxymodel.cpp

```diff
@@ -183,5 +183,21 @@ bool OpportunityFilterProxyModel::lessThan(const QModelIndex &left, const QModel
             return leftDt < rightDt;
         }
     }
-    return FilterProxyModel::lessThan(left, right);
+
+    if (FilterProxyModel::lessThan(left, right)) {
```

Comment:
It is quite surprising that this isn't symmetric (if lessThan(right, left) then we can just go ahead and return true). 
It means calling lessThan() twice, but that's basically what this code does below when fetching the values "l" and "r", so it wouldn't really be worse. I think it would be better, to avoid subtle bugs due to any difference between QSFPM::lessThan() and our equality test below.

After testing lessThan() both ways, you know that we're in the equality case and you can then fallback to creation date --- if the sort column isn't the creation date in the first place, otherwise the fallback makes no sense (that check is missing right now)